test/unit_tests/augmentation/test_transforms.py

```diff
@@ -631,6 +636,60 @@ def test_mixup_transform(rng_seed, random_batch, alpha, beta_per_sample):
 ]
 
 
+@pytest.fixture()
+def dataset():
+    return concat_windows_dataset(concat_ds_targets())
+
+
+@pytest.mark.parametrize("probability", [0, 0.5, 1])
+def test_segmentation_reconstruction_with_EEGClassifier(dataset, probability):
+
+    transform = SegmentationReconstruction(probability=probability)
+
+    model = ShallowFBCSPNet(n_times=750, n_chans=4, n_classes=2)
+
+    clf = EEGClassifier(
+        module=model,
+        optimizer=torch.optim.Adam,
+        iterator_train=AugmentedDataLoader,
+        iterator_train__transforms=transform,
+        batch_size=5,
+        max_epochs=1,
+        classes=[0, 1],
+        train_split=predefined_split(dataset),
+        verbose=0,
+    )
+
+    _ = clf.fit(X=dataset)
```

Comment:
I wonder whether this test is worth it. The idea of testing that the training pipeline works using augmentations is appealing, but costs a lot in CI time, which we would like to keep low. If we wanted to do this, we could not write this kind of test for every augmentation, so we would probably want to do something generic (not specific to SegmentationReconsruction augmentation). As for now, I think the test at line 710 is enough to check pipeline compliance and would remove this one.

What do you think @bruAristimunha ?

Reply:
Hi @cedricrommel! =)

I created this extra test to catch an error not being seen with the test on line 710. The mask integration was not working correctly. @gustavohenriquesr solved here: https://github.com/braindecode/braindecode/pull/608/commits/def88c59891b63c936aefc18503b327b93f3cb94

Regarding creating this style of test integration, I understand that it increases CI time a little, but since [573](https://github.com/braindecode/braindecode/pull/573) and [571](https://github.com/braindecode/braindecode/pull/571), we have been adopting this approach, as some models were not working in our braindecode contract. @agramfort made the same point you made in the past, but as the benefits are necessary, it's okay to have this integration test. Alex comments [here](https://github.com/braindecode/braindecode/pull/573#discussion_r1491712651).

Regarding this specific test, it is outside our top 10 tests that take the most time; I suggest keeping it and modifying it to be generic for all augmentations if you want.

I spent time optimizing testing and documentation during the last sprint. If necessary, the secret will be to optimize the FakeData loader more.